File: backend/TaskTree/Models/Project.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace TaskTree.Models
{
    public class Project : BaseEntity
    {
        [Required]
        public string? Name { get; set; }
        public string? Description { get; set; }
        public Task? Root { get; set; }
        public double? Progress { get; set; }
        [Required]
        public User? User { get; set; }
    }
}

File: frontend/src/types/Project.ts
import Task, { BlankTask } from './Task';

export default interface Project {
  id?: number;
  name: string;
  description?: string;
  root?: Task;
  progress: number;
  updatedAt?: Date;
}

export const BlankProject: Project = {
  id: 0,
  name: '',
  description: '',
  progress: 0,
  root: BlankTask
};
